vhost-device-gpu/src/backend/gfxstream.rs

```diff
@@ -0,0 +1,1163 @@
+        Ok(OkNoData)
+    }
+
+    fn submit_command(
+        &mut self,
+        ctx_id: u32,
+        commands: &mut [u8],
+        fence_ids: &[u64],
+    ) -> VirtioGpuResult {
+        Self::with_rutabaga(|rutabaga| rutabaga.submit_command(ctx_id, commands, fence_ids))?;
+        Ok(OkNoData)
+    }
+
+    fn create_fence(&mut self, rutabaga_fence: RutabagaFence) -> VirtioGpuResult {
+        Self::with_rutabaga(|rutabaga| rutabaga.create_fence(rutabaga_fence))?;
+        Ok(OkNoData)
+    }
+
+    fn process_fence(
+        &mut self,
+        ring: VirtioGpuRing,
+        fence_id: u64,
+        desc_index: u16,
+        len: u32,
+    ) -> bool {
+        common::common_process_fence(&self.fence_state, ring, fence_id, desc_index, len)
+    }
+
+    fn get_event_poll_fd(&self) -> Option<EventFd> {
+        Self::with_rutabaga(|rutabaga| {
+            rutabaga.poll_descriptor().map(|fd| {
+                // SAFETY: Safe, the fd should be valid, because Rutabaga guarantees it.
+                // into_raw_descriptor() returns a RawFd and makes sure SafeDescriptor::drop
+                // doesn't run.
+                unsafe { EventFd::from_raw_fd(fd.into_raw_descriptor()) }
+            })
+        })
+    }
+
+    fn event_poll(&self) {
+        Self::with_rutabaga(|rutabaga| {
+            rutabaga.event_poll();
+        });
+    }
+
+    fn force_ctx_0(&self) {
+        Self::with_rutabaga(|rutabaga| {
+            rutabaga.force_ctx_0();
+        });
+    }
+
+    fn display_info(&self) -> VirtioGpuResult {
+        common::common_display_info(&self.gpu_backend)
+    }
+
+    fn get_edid(&self, edid_req: VhostUserGpuEdidRequest) -> VirtioGpuResult {
+        common::common_get_edid(&self.gpu_backend, edid_req)
+    }
+
+    fn set_scanout(
+        &mut self,
+        scanout_id: u32,
+        resource_id: u32,
+        rect: virtio_gpu_rect,
+    ) -> VirtioGpuResult {
+        let scanout_idx = scanout_id as usize;
+        if resource_id == 0 {
+            common_set_scanout_disable(&mut self.scanouts, scanout_idx);
+
+            self.gpu_backend
+                .set_scanout(&VhostUserGpuScanout {
+                    scanout_id,
+                    width: 0,
+                    height: 0,
+                })
+                .map_err(|e| {
+                    error!("Failed to disable scanout: {e:?}");
+                    ErrUnspec
+                })?;
+
+            return Ok(OkNoData);
+        }
+
+        // If there was a different resource previously associated with this scanout,
+        // disable the scanout on that old resource
+        if let Some(old_scanout) = &self.scanouts[scanout_idx] {
+            let old_resource_id = old_scanout.resource_id;
+            if old_resource_id != resource_id {
+                if let Some(old_resource) = self.resources.get_mut(&old_resource_id) {
+                    old_resource.scanouts.disable(scanout_id);
+                }
+            }
+        }
```

Comment:
Why this was added in the last forced push?

Reply:
I realized compiling only the individual backends like for gfxstream had some unused variables and I noticed this bit was also important, so I added it to match what was existing in virgl backend.